streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -227,6 +255,8 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
       case DeregisterReceiver(streamId, message, error) =>
         deregisterReceiver(streamId, message, error)
         context.reply(true)
+      case ScheduleReceiver(streamId) =>
```

Comment:
I think this messaging should be done differently to align the messaging style with the other messages. It does not make sense for the receiver to fetch all the allowed locations that the receiver is allowed to start. Rather it should be 
`ShouldStartReceiver`, which returns true or false, and accordingly the receiver is started by the `ReceiverSupervisor`.